User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs

```diff
@@ -0,0 +1,388 @@
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => new CollectionSortHandler<double>(null, writer, parser));
+        }
+
+        /// <summary>
+        /// Collections the sort handler collection sort handler pass null value in constructor throw argument null exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_CollectionSortHandler_PassNullWriterInConstructor_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var parser = new StubIDataParser<double>();
+            var reader = new StubDataReader(new StubIValidator());
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => new CollectionSortHandler<double>(reader, null, parser));
+        }
+
+        /// <summary>
+        /// Tests Execute if left empty unsorted collection it will throw exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_Execute_LeftEmptyUnsortedCollection_ThrowException()
+        {
+            using (ShimsContext.Create())
+            {
+                //
+                // Arrange.
+                //
+                var parser = new StubIDataParser<double>();
+                var reader = new ShimDataReader();
+                var writer = new StubDataWriter<double>();
+                CollectionSortHandler<double> collectionSortHandler;
+
+                //
+                // Act.
+                //
+                collectionSortHandler = new CollectionSortHandler<double>(reader, writer, parser);
+
+                //
+                // Assert.
+                //
+                Assert.ThrowsException<Exception>(() => collectionSortHandler.Execute());
+            }
+        }
+
+        /// <summary>
+        /// Tests Execute if pass InsertionSort as SortType it will get sorted collection according to that type.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_Execute_PassInsertionSortType_GetSortedCollection()
+        {
+            using (ShimsContext.Create())
+            {
+                //
+                // Arrange.
+                //
+                var expectedCollection = new double[] { 1.5, 4, 11, 56, 166 };
+                var sorterType = SortType.InsertionSort;
+                var path = @"D:\collectionToRead.txt";
+                var parser = new StubIDataParser<double>()
+                {
+                    ConvertDataString = (str) => new double[] { 166, 11, 56, 4, 1.5 }
+                };
+                var reader = new ShimDataReader();
+                var writer = new StubDataWriter<double>();
+                CollectionSortHandler<double> collectionSortHandler;
+                bool result;
```

Comment:
Please renamae result to something more understandable, like isCollectionSorted.